### ISSUE TITLE
reproduction of kseg

### ISSUE THREAD

**Issue by DopamineLcy** [lin-tianyu/Stable-Diffusion-Seg#17]
Thank you for your impressive work!
I tried reproducing your results of kseg strictly according to your README without changing any config, however, I surprisingly found the results of kseg notably lower than the reported ones.

Is there any potential mistake that leads to the inconsistency?
Thank you again and looking forward to your reply.

![image](https://github.com/user-attachments/assets/295198bb-8ef5-424e-aa88-3ca99b38b4f4)
![image](https://github.com/user-attachments/assets/17146fc4-35a0-48b4-b1c2-10edb695f6f2)

**Comment by DopamineLcy** [comment]
Would you mind sharing the data splits of keg, I'm not sure whether the stochasticity of glob leads to different data plits.

**Comment by lin-tianyu** [comment]
Hi, thanks for your interest. Unfortunately, I don't have a data splits of kseg, since the data split I use is in the dataloader file `kseg.py` with the `_parse_data_list` method. I am not sure whether this is a problem from glob because my previous experiment results seems consistent.

Have you checked the tensorboard of the training log? What does it look like? Maybe you can get more insight from the tensorboard.

If you still cannot fix the problem, I will try reomplement it when I'm free. Thanks!

**Comment by tara0907** [comment]
> 感谢您令人印象深刻的工作！我尝试严格按照您的自述文件重现您的 kseg 结果，而不更改任何配置，但是，我惊讶地发现 kseg 的结果明显低于报告的结果。
> 
> 是否有任何可能导致不一致的潜在错误？再次感谢您，期待您的回复。
> 
> ![image](https://github.com/user-attachments/assets/295198bb-8ef5-424e-aa88-3ca99b38b4f4) ![image](https://github.com/user-attachments/assets/17146fc4-35a0-48b4-b1c2-10edb695f6f2)

Hello, my results on the KSEG dataset are far worse than the original results (I ran about 500 epochs). Did you solve this problem?

<img width="687" height="404" alt="Image" src="https://github.com/user-attachments/assets/4e6d7e64-664e-4bba-9787-b1139d63244d" />

**Comment by tara0907** [comment]
> Hi, thanks for your interest. Unfortunately, I don't have a data splits of kseg, since the data split I use is in the dataloader file `kseg.py` with the `_parse_data_list` method. I am not sure whether this is a problem from glob because my previous experiment results seems consistent.
> 
> Have you checked the tensorboard of the training log? What does it look like? Maybe you can get more insight from the tensorboard.
> 
> If you still cannot fix the problem, I will try reomplement it when I'm free. Thanks!

Dear author, thank you for sharing this excellent work. I'd like to know how many epochs your code ran in total. I can't find any setting for the number of epochs in the project documentation. Regarding the discrepancy between the KSEG dataset results and your paper, is the number of epochs significantly impacting the results? Also, I'm training on an A100 with an AI00 80G processor. Does increasing the batch size to 8 affect the final results? I look forward to your response!

**Comment by lin-tianyu** [comment]
For one, I always look at `steps` rather than `epoch`. And 32k steps are not so much. There is a chance that the model have not converged yet. Please checkout tensorboard logs to see the curve, that'll help.

Also, some people have performance issue on CVC dataset as well. After thorough experiments and discussion with @WhoamiV (https://github.com/lin-tianyu/Stable-Diffusion-Seg/issues/36) and @12SA3 (https://github.com/lin-tianyu/Stable-Diffusion-Seg/issues/32), we believe these performance drop is a random seed issue: **same random seed may perform different behavior on different machine.** For example, I have rerun CVC experiment and get normal performance, but @WhoamiV running the same code got >5% performance drop.

If this is the case, there is nothing I can do to help. I can only suggest changing random seed. Best of luck!

**Comment by lin-tianyu** [comment]
I think I mentioned 100k steps in the paper. Small datasets like KSeg should use around 50~60k steps to converge just fine.

If you by any chance have V100 or A6000 GPU, I would recommand using them, since I have produced normal results on these machine. I don't know if this is the problem here.

Increasing batch size sould help. But I doubt this to be reason of the performance drop.

**Comment by tara0907** [comment]
> I think I mentioned 100k steps in the paper. Small datasets like KSeg should use around 50~60k steps to converge just fine.
> 
> If you by any chance have V100 or A6000 GPU, I would recommand using them, since I have produced normal results on these machine. I don't know if this is the problem here.
> 
> Increasing batch size sould help. But I doubt this to be reason of the performance drop.

Thank you very much for your reply despite your busy schedule. I will try again according to your suggestions. Thanks again!

**Comment by xiao-qian-a** [comment]
> > I think I mentioned 100k steps in the paper. Small datasets like KSeg should use around 50~60k steps to converge just fine.
> > If you by any chance have V100 or A6000 GPU, I would recommand using them, since I have produced normal results on these machine. I don't know if this is the problem here.
> > Increasing batch size sould help. But I doubt this to be reason of the performance drop.
> 
> Thank you very much for your reply despite your busy schedule. I will try again according to your suggestions. Thanks again!

Hello, I encountered the same issue as you when reproducing the Kvasir dataset - the results for Kvasir-SEG are significantly lower than those reported in the paper. May I ask if you have resolved this? Did you get closer to the reported results after changing your hardware setup?
I have also reviewed the related discussions. After checking, my cond_stage_trainable is set to True. I am using an RTX 4090 D graphics card.
Here are my reproduction results:

<img width="1461" height="665" alt="Image" src="https://github.com/user-attachments/assets/6754637a-dfe9-4eae-a9b2-da61079dd0af" />

I sincerely hope you can find the time to reply despite your busy schedule.

**Comment by lin-tianyu** [comment]
Long story short, many people who have this issue and myself believe that this is an issue caused by a random dataset split. 

But the bad news is that, this cannot be resolved even if we set the same random seed. It should be some kind of hardware issue (My experiments are run on a V100 GPU).

So a simple way to fix this is to try different random seeds. Hope this helps.

Also if you don't necessarily need to run this dataset, please try bigger ones. A bigger dataset (e.g., BTCV) brings much more stable results according to my experience. It is not worth it to waste too much time on this.

Best,
Tianyu
